File: Queue/Python/queueClass.py
class Queue:
    def __init__(self) -> None:
        self.queue = []

    def push(self, data):
        self.queue.append(data)

    def pop(self):
        self.queue.pop(0)

    def front(self):
        return self.queue[0]

    def printQueue(self):
        print(self.queue)

    def size(self):
        return len(self.queue)

    def empty(self):
        return len(self.queue) == 0


q = Queue()

q.push('a')
q.push('b')
q.push('c')

print(q.front())
q.printQueue()

q.pop()
q.pop()
q.pop()


print(q.empty())

File: Queue/C++/QueueClass.cpp
#include <iostream>
#include <vector>
using namespace std;

class Queue
{
public:
    vector<int> v;

    Queue()
    {
        this->v = {};
    };

    void push(int val)
    {
        this->v.push_back(val);
    };
    void pop()
    {
        if (!this->v.empty())
            this->v.erase(this->v.begin());
        else
            cout << "Queue is empty" << endl;
    };
    int front()
    {
        return this->v[0];
    }
};

int main()
{
    Queue q;
    q.push(1);
    q.push(2);
    q.push(3);
    q.pop();
    cout << q.front() << endl;
    return 1;
}